google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -120,65 +123,89 @@
 
   public static EnhancedBigtableStub create(EnhancedBigtableStubSettings settings)
       throws IOException {
-    ClientContext clientContext = ClientContext.create(settings);
+    settings = finalizeSettings(settings, Tags.getTagger(), Stats.getStatsRecorder());
 
-    return new EnhancedBigtableStub(
-        settings, clientContext, Tags.getTagger(), Stats.getStatsRecorder());
+    return new EnhancedBigtableStub(settings, ClientContext.create(settings));
   }
 
-  @InternalApi("Visible for testing")
-  public EnhancedBigtableStub(
-      EnhancedBigtableStubSettings settings,
-      ClientContext clientContext,
-      Tagger tagger,
-      StatsRecorder statsRecorder) {
-    this.settings = settings;
+  public static EnhancedBigtableStubSettings finalizeSettings(
```

Comment:
this is a little confusing to me. Why did we separate these pieces out?

Reply:
Not sure which specific pieces you are referring to, but I'll try to give the background to the current structure.

There a couple of phases of client construction:
1. user uses the builder to configure their settings. At this point the user owns the object so modifying it in place during build() is very surprising
2. stub creation: at this point the user passed us the settings object. This is the correct place to augment the settings as the changes won't be visible in the user object

Previously step 2 was handled when we created ClientContext, which is a manifestation of the user settings.  Which is the correct place to do this. Unfortunately for this new feature, the primer needs to be injected before the ClientContext is created (because ClientContext instantiates the grpc connections). So the settings need to be augmented before the ClientContext is created. So ClientContext.create(settings).toBuilder().augment() was changed to ClientContext.create(settings.toBuilder().augment()). 

At this point the logic of patching settings got pretty hefty and fairly distinct from callable chain creation, so it was extracted to a helper method.

Reply:
Thanks! By 'pieces' I was mostly referring to the channel priming and the tracing - wasn't quite sure how they went together. Do you think it would be clearer if we had a `finalizeSettings` which delegates to each sub instantiation (channel priming, tracing)?

Reply:
I'd like to leave it as is, we can factor those out when we get a 3rd feature

Reply:
ok, then for now I think a comment would be helpful

Reply:
done